### PR TITLE
Amc docs

### PR DESCRIPTION
**Pull request by alexeilebedev**

### CODE REVIEW

File: data/dmmeta/ctype.ssim [hunk 1]
```diff
@@ -724,119 +724,119 @@ dmmeta.ctype  ctype:dev.Timefmt  comment:"Time formats supported by orgfile"
...
-dmmeta.ctype  ctype:dmmeta.Basepool  comment:""
-dmmeta.ctype  ctype:dmmeta.Bitfld  comment:""
+dmmeta.ctype  ctype:dmmeta.Anonfld  comment:"Omit field name where possible (command line, enums, constants)"
+dmmeta.ctype  ctype:dmmeta.Argvtype  comment:"Customize parsing of command lines (rarely used)"
+dmmeta.ctype  ctype:dmmeta.Basepool  comment:"Specify pool to be used for allocating elements of a type"
+dmmeta.ctype  ctype:dmmeta.Bitfld  comment:"Specify offset/width of a bitfield within another field"
 dmmeta.ctype  ctype:dmmeta.Buftype  comment:""
 dmmeta.ctype  ctype:dmmeta.BuftypeId  comment:""
-dmmeta.ctype  ctype:dmmeta.Cafter  comment:""
+dmmeta.ctype  ctype:dmmeta.Cafter  comment:"Force ordering of unrelated ctypes during topological sort"
```
Comment (polovinkin-alexey) [inline data/dmmeta/ctype.ssim:733]: not sure that I understand this

File: data/dmmeta/ctype.ssim [hunk 2]
```diff
@@ -724,119 +724,119 @@ dmmeta.ctype  ctype:dev.Timefmt  comment:"Time formats supported by orgfile"
...
 dmmeta.ctype  ctype:dmmeta.Cascdel  comment:"Request cascading delete of referenced rows"
 dmmeta.ctype  ctype:dmmeta.Ccmp  comment:"Generate hash function"
-dmmeta.ctype  ctype:dmmeta.Cdecl  comment:""
-dmmeta.ctype  ctype:dmmeta.Cdflt  comment:""
+dmmeta.ctype  ctype:dmmeta.Cdecl  comment:"Request forward-declaration of ctype"
```
Comment (polovinkin-alexey) [inline data/dmmeta/ctype.ssim:736]: why it is needed? i think all types shall be forward-declard

File: data/dmmeta/ctype.ssim [hunk 3]
```diff
@@ -724,119 +724,119 @@ dmmeta.ctype  ctype:dev.Timefmt  comment:"Time formats supported by orgfile"
...
+dmmeta.ctype  ctype:dmmeta.Cdflt  comment:"Specify default value for single-value types that lack fields"
 dmmeta.ctype  ctype:dmmeta.Cextern  comment:"Externally defined ctype (a struct from system header, or from a c++ library)"
-dmmeta.ctype  ctype:dmmeta.Cfmt  comment:""
+dmmeta.ctype  ctype:dmmeta.Cfmt  comment:"Specify options for printing/reading ctypes into multiple formats"
 dmmeta.ctype  ctype:dmmeta.Cget  comment:"Generate state functions for these ctypes"
-dmmeta.ctype  ctype:dmmeta.Charset  comment:""
+dmmeta.ctype  ctype:dmmeta.Charset  comment:"Generate char membership function (as bitset or inline calculation)"
```
Comment (polovinkin-alexey) [inline data/dmmeta/ctype.ssim:741]: I do not understand it at all

File: data/dmmeta/ctype.ssim [hunk 4]
```diff
@@ -724,119 +724,119 @@ dmmeta.ctype  ctype:dev.Timefmt  comment:"Time formats supported by orgfile"
...
 dmmeta.ctype  ctype:dmmeta.Chash  comment:"Generate hash function"
 dmmeta.ctype  ctype:dmmeta.CppExpr  comment:""
 dmmeta.ctype  ctype:dmmeta.Cppfunc  comment:"Value of field provided by this expression"
 dmmeta.ctype  ctype:dmmeta.Cppkeyword  comment:""
-dmmeta.ctype  ctype:dmmeta.Cpptype  comment:""
-dmmeta.ctype  ctype:dmmeta.Csize  comment:""
-dmmeta.ctype  ctype:dmmeta.Cstr  comment:""
-dmmeta.ctype  ctype:dmmeta.Ctype  comment:"C structure"
+dmmeta.ctype  ctype:dmmeta.Cpptype  comment:"Specify that a type should be passed by value, or behaves like data"
```
Comment (polovinkin-alexey) [inline data/dmmeta/ctype.ssim:746]: it s only a single field

File: data/dmmeta/ctype.ssim [hunk 5]
```diff
@@ -724,119 +724,119 @@ dmmeta.ctype  ctype:dev.Timefmt  comment:"Time formats supported by orgfile"
...
+dmmeta.ctype  ctype:dmmeta.Csize  comment:"Specify field size/alignment for built-in C++ types"
```
Comment (polovinkin-alexey) [inline data/dmmeta/ctype.ssim:747]: size only?

File: data/dmmeta/ctype.ssim [hunk 6]
```diff
@@ -724,119 +724,119 @@ dmmeta.ctype  ctype:dev.Timefmt  comment:"Time formats supported by orgfile"
...
+dmmeta.ctype  ctype:dmmeta.Cstr  comment:"Specify that type behaves like a string"
+dmmeta.ctype  ctype:dmmeta.Ctype  comment:"Generate C struct"
```
Comment (polovinkin-alexey) [inline data/dmmeta/ctype.ssim:749]: it is a conceptual thing, not onlt c-struct

File: data/dmmeta/ctype.ssim [hunk 7]
```diff
@@ -724,119 +724,119 @@ dmmeta.ctype  ctype:dev.Timefmt  comment:"Time formats supported by orgfile"
...
+dmmeta.ctype  ctype:dmmeta.Nsx  comment:"Control code-generation and exception handling options for process/library"
+dmmeta.ctype  ctype:dmmeta.Numstr  comment:"Add functions to read numbers out of a string field"
 dmmeta.ctype  ctype:dmmeta.Pack  comment:"Request byte-packing of structure fields"
-dmmeta.ctype  ctype:dmmeta.Pmaskfld  comment:""
-dmmeta.ctype  ctype:dmmeta.Pnew  comment:"placement new"
+dmmeta.ctype  ctype:dmmeta.Pmaskfld  comment:"Specify which fields holds presence mask bits (one for each field in struct)"
+dmmeta.ctype  ctype:dmmeta.Pnew  comment:"Generate custom constructor (placement new), for use with binary protocols"
 dmmeta.ctype  ctype:dmmeta.Printfmt  comment:""
-dmmeta.ctype  ctype:dmmeta.Ptrary  comment:""
+dmmeta.ctype  ctype:dmmeta.Ptrary  comment:"Annotate Ptrary field"
```
Comment (polovinkin-alexey) [inline data/dmmeta/ctype.ssim:818]: just annotate?

File: txt/amc.md [hunk 8]
```diff
@@ -20,27 +144,55 @@ Functions to convert any struct to/from a string or a bash command line; Enum su
...
 where you have to "plug in" anything. User always controls what functions will be called
 (dispatches and steps are an exception, but they are also explicitly controllable).
-`amc` is a tool for constructing software based on your specifications.
+`amc` is a tool for constructing robust software based on your specifications.
+
+The authors have spent many years using template-based libraries and
+class hierarchies, and found them to be unmaintainable in the long
+run.  amc represents a step forward by stepping away from these
+concepts; The templates, which are a form of compile-time code
+generation, are fully superceded by explicit code generation, which
```
Comment (polovinkin-alexey) [inline txt/amc.md:174]: supersesed

File: txt/amc.md [hunk 9]
```diff
@@ -20,27 +144,55 @@ Functions to convert any struct to/from a string or a bash command line; Enum su
...
+can perform deductive reaasoning on closed sets, something that's not
```
Comment (polovinkin-alexey) [inline txt/amc.md:175]: reasoning

**Review by polovinkin-alexey — APPROVED** [review]

File: data/dmmeta/ctype.ssim (hunk 2 above)
Comment (alexeilebedev) [reply data/dmmeta/ctype.ssim:736]: deleted dmmeta.cdecl

File: data/dmmeta/ctype.ssim (hunk 3 above)
Comment (alexeilebedev) [reply data/dmmeta/ctype.ssim:741]: rewritten

**Pull request merged**